main.rb

```diff
@@ -0,0 +1,120 @@
+module Enumerable
+  # each
+  def my_each
+    each do |i|
+      yield i
+    end
+  end
+
+  # each_with_index
+  def my_each_with_index
+    result = map do |element, index|
+      yield "#{index}: #{element} "
+    end
+    result
+  end
+
+  # select
+  def my_select
+    result = []
+    each do |i|
+      result << i if yield(i)
+    end
+    result
+  end
+
+  # all
+  def my_all?
+    my_each do |_elememt|
+      return false unless yield(element)
+    end
+    true
+  end
+
+  # any
+  def my_any?
+    my_each do |_elememt|
+      return true if yield(element)
+    end
+    false
+  end
+
+  # negated
+  def my_none?
+    my_each do |element|
+      yield(element)
+    end
+  end
```

Comment:
- [ ] `my_none?` returns true if the block never returns true for all elements
`[4, 5, 9, 8, 10].my_none? { |num| num > 12}` return true

- [ ] `my_none?` when no block or argument is given returns true only if none of the collection members is true
`[false, nil].my_none?` returns true

- [ ]  my_none? when a class is passed as an argument returns true if none of the collection is a member of such class
`[1, 3, 42, 9].none?(Float)` returns true

- [ ] `my_none?` when a Regex is passed as an argument returns true only if none of the collection matches the Regex
`%w{ant bear cat}.none?(/d/)` returns true

- [ ] When a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
`p [1,7,3,9,2,3].my_none?(5)` should return true